draft-ietf-tls-ctls.md

```diff
@@ -361,11 +408,55 @@ mistaken for compressed one and erroneously decompressed.  For X.509, it is
 sufficient for the first byte of the compressed value (key) to have a value
 other than 0x30, since every X.509 certificate starts with this byte.
 
+### Static Vector compression
+
+When the cTLS template implies that a variable-length vector ({{!RFC8446, Section 3.4}}) has a fixed number of elements, that vector's length prefix is omitted.
+For example, suppose that the cTLS template is:
+
+~~~JSON
+{
+  "version": 772,
+  "dh_group": "x25519",
+  "client_hello_extensions": {
+    "expected_extensions": ["key_share"],
+    "allow_additional": false
+  }
+}
+~~~
+
+Then, the following structure:
+
+~~~
+   28                 // length(extensions)
+   33 26              // extension_type = KeyShare
+     0024             // length(client_shares)
+       001d           // KeyShareEntry.group
+       0020           // length(KeyShareEntry.key_exchange)
+         a690...af948 // KeyShareEntry.key_exchange
+~~~
+
+is compressed down to:
+
+~~~
+   a690...af948 // KeyShareEntry.key_exchange
+~~~
+
+according to the following rationale:
+:
+* The length of the `key_exchange` is omitted because the "x25519" key share has a fixed size (32 bytes).
+* `KeyShareEntry.group` is omitted because it is specified by `dh_group`
+* The length of `client_shares` is omitted because the use of `dh_group` implies that
+there can only be one `dh_group`.
+* `extension_type` is omitted because it is specified by `expected_extensions`
+* The length of `extensions` is omitted because `allow_additional` is false, the number of items in `extensions` (i.e., 1) is known in advance.
+
+The only exception to this rule is `ClientHello.profile_id`, which is processed before the profile is known.
```

Comment:
How clever will one have to be to get this right?

Reply:
My goal here was mostly to capture a behavior that was implied by some examples but not described anywhere in the text.  I've attempted to make this rule more explicit, to avoid any tricky reasoning being required.